docs/ory_permission_language_spec.md

```diff
@@ -0,0 +1,303 @@
+[Authzed Schema Language](https://docs.authzed.com/reference/schema-lang) and
+[Auth0 FGA](https://docs.fga.dev/). Most permissions are defined by normal
+developers who typically are most familiar with Web technologies like JavaScript
+or Typescript. We see the need for a developer-friendly configuration language
+for permissions that has such a small learning curve that most developers can
+understood and use it with close to no effort. We therefore chose to define our
+permissions configuration language as a subset of the most common
+general-purpose programming language: JavaScript/TypeScript.
+
+The Ory Permission Language is a syntactical subset of TypeScript. Along with
+type definitions for the syntax elements of the language (such as `Namespace` or
+`Context`), users can get context help from their IDE while writing the
+configuration.
+
+## Notation
+
+The syntax is specified using Extended Backus-Naur Form (EBNF):
+
+```ebnf
+Production  = production_name "=" [ Expression ] "." .
+Expression  = Alternative { "|" Alternative } .
+Alternative = Term { Term } .
+Term        = production_name | token [ "…" token ] | Group | Option | Repetition .
+Group       = "(" Expression ")" .
+Option      = "[" Expression "]" .
+Repetition  = "{" Expression "}" .
+```
+
+Productions are expressions constructed from terms and the following operators,
+in increasing precedence:
+
+```ebnf
+|   alternation
+()  grouping
+[]  option (0 or 1 times)
+{}  repetition (0 to n times)
+```
+
+Lower-case production names are used to identify lexical tokens. Non-terminals
+are in CamelCase. Lexical tokens are enclosed in double quotes `""` or single
+quotes `''`.
+
+The form `a … b` represents the set of characters from a through b as
+alternatives. The horizontal ellipsis `…` is also used elsewhere in the spec to
+informally denote various enumerations or code snippets that are not further
+specified.
+
+## Configuraton text representation
+
+The configuration is encoded in UTF-8.
+
+## Lexical elements
+
+### Comments
+
+1. Line comments start with the character sequence `//` and stop at the end of
+   the line.
+2. General comments start with the character sequence `/*` and stop with the
+   first subsequent character sequence `*/`.
+3. Documentation comments start with the character sequence `/**` and stop with
+   the first subsequent character sequence `*/`.
+
+### Identifiers
+
+Identifiers name program entities such as variables and types. An identifier is
+a sequence of one or more letters and digits. The first character in an
+identifier must be a letter.
+
+```ebnf
+identifier = letter { letter | digit } .
+digit      = "0" … "9" .
+letter     = "A" … "Z" | "a" … "z" | "_" .
+```
+
+### String literals
+
+String literals represent string constants as sequences of characters.
+
+```ebnf
+string_lit    = single_quoted | double_quoted .
+single_quoted = "'" { letter | digit } "'" .
+double_quoted = '"' { letter | digit } '"' .
```

Comment:
Can strings contain spaces too?

Reply:
Ah, now that we removed metadata we don't have to have this in the spec at all!